src/Sulu/Bundle/WebsiteBundle/Twig/Content/ContentPathTwigExtension.php

```diff
@@ -66,41 +58,55 @@ public function getFunctions()
     /**
      * {@inheritdoc}
      */
-    public function getContentPath($url, $webspaceKey = null, $locale = null, $domain = null, $scheme = null)
+    public function getContentPath($route, $webspaceKey = null, $locale = null, $domain = null, $scheme = null)
     {
         if (!$this->requestAnalyzer) {
-            return $url;
+            return $route;
         }
 
-        $domain = $domain ?: $this->requestAnalyzer->getAttribute('host');
         $scheme = $scheme ?: $this->requestAnalyzer->getAttribute('scheme');
         $locale = $locale ?: $this->requestAnalyzer->getCurrentLocalization()->getLocale();
 
         if ($webspaceKey !== null) {
-            if (!$this->webspaceManager->findWebspaceByKey($webspaceKey)->hasDomain($domain, $this->environment)) {
+            if (null !== $domain
+                && !$this->webspaceManager->findWebspaceByKey($webspaceKey)->hasDomain($domain, $this->environment)
+            ) {
                 $domain = null;
             }
 
             $url = $this->webspaceManager->findUrlByResourceLocator(
-                $url,
+                $route,
                 $this->environment,
                 $locale,
                 $webspaceKey,
-                $domain,
+                $domain ?: $this->requestAnalyzer->getAttribute('host'),
                 $scheme
             );
-        } elseif (strpos($url, '/') === 0) {
-            $url = $this->webspaceManager->findUrlByResourceLocator(
-                $url,
+
+            if (!$url && !$domain) {
+                $url = $this->webspaceManager->findUrlByResourceLocator(
+                    $route,
+                    $this->environment,
+                    $locale,
+                    $webspaceKey,
+                    null,
+                    $scheme
+                );
+            }
+
+            return $url;
+        } elseif (strpos($route, '/') === 0) {
```

Comment:
Is this if really necessary? Due to your changes this seems to be exactly the same call to `findUrlByResourceLocator` as in https://github.com/sulu/sulu/pull/2363/files#diff-896fde80dda7a566a31bc99704eaab23R77